Data/ByteString/Short/Internal.hs

```diff
@@ -446,6 +578,849 @@ concat sbss =
       copyByteArray (asBA src) 0 dst off len
       copy dst (off + len) sbss
 
+-- ---------------------------------------------------------------------
+-- Basic interface
+
+infixr 5 `cons` --same as list (:)
+infixl 5 `snoc`
+
+-- | /O(n)/ Append a byte to the end of a 'ShortByteString'
+-- 
+-- Note: copies the entire byte array
+--
+-- @since 0.11.3.0
+snoc :: ShortByteString -> Word8 -> ShortByteString
+snoc = \sbs c -> let l = length sbs
+                     nl = l + 1
+  in create nl $ \mba -> do
+      copyByteArray (asBA sbs) 0 mba 0 l
+      writeWord8Array mba l c
+{-# INLINE snoc #-}
+
+-- | /O(n)/ 'cons' is analogous to (:) for lists.
+--
+-- Note: copies the entire byte array
+--
+-- @since 0.11.3.0
+cons :: Word8 -> ShortByteString -> ShortByteString
+cons c = \sbs -> let l = length sbs
+                     nl = l + 1
+  in create nl $ \mba -> do
+      writeWord8Array mba 0 c
+      copyByteArray (asBA sbs) 0 mba 1 l
+{-# INLINE cons #-}
+
+-- | /O(1)/ Extract the last element of a ShortByteString, which must be finite and non-empty.
+-- An exception will be thrown in the case of an empty ShortByteString.
+--
+-- @since 0.11.3.0
+last :: HasCallStack => ShortByteString -> Word8
+last = \sbs -> case null sbs of
+  True -> error "empty ShortByteString"
+  False -> indexWord8Array (asBA sbs) (length sbs - 1)
+{-# INLINE last #-}
+
+-- | /O(n)/ Extract the elements after the head of a ShortByteString, which must be non-empty.
+-- An exception will be thrown in the case of an empty ShortByteString.
+--
+-- Note: copies the entire byte array
+--
+-- @since 0.11.3.0
+tail :: HasCallStack => ShortByteString -> ShortByteString
+tail = \sbs -> 
+  let l = length sbs
+      nl = l - 1
+  in case null sbs of
+      True -> error "empty ShortByteString"
+      False -> create nl $ \mba -> copyByteArray (asBA sbs) 1 mba 0 nl
+{-# INLINE tail #-}
+
+-- | /O(n)/ Extract the head and tail of a ByteString, returning Nothing
+-- if it is empty.
+uncons :: ShortByteString -> Maybe (Word8, ShortByteString)
+uncons sbs =
```

Comment:
This should inline better:

```suggestion
uncons = \sbs ->
```

Please use the same style in other functions that take a `ShortByteString` as the last parameter, e.g. `unsnoc`, `takeWhile` etc.

Reply:
Actually, there's no reason to limit this to functions that have a `ShortByteString` as the last parameter. E.g. in `intercalate`, the `[ShortByteString]` argument should be bound in a lambda.